fastcrypto-vdf/src/class_group/discriminant.rs

```diff
@@ -41,18 +55,25 @@ impl Discriminant {
         &self.0
     }
 
-    /// Check the primality of this discriminant and return an error if it is not prime.
-    pub fn check_primality(&self) -> FastCryptoResult<()> {
-        match is_probable_prime(
-            &self
-                .0
-                .abs()
-                .to_biguint()
-                .expect("Absolute value is non-negative"),
-        ) {
-            true => Ok(()),
-            false => Err(InvalidInput),
-        }
+    /// Create a discriminant from the given bytes. The bytes must be an BCS encoding of the
+    /// twos-complement big-endian encoding of the discriminant. Returns an error if the encoding is
+    /// invalid.
+    ///
+    /// It is assumed that the discriminant is a negative prime which is 1 mod 8. If this is not the
```

Comment:
Do we ensure the real use case on the blockchain cannot panic? I guess we don't provide access to that, right?

Reply:
No, this is only called with a fixed, safe discriminant. Maybe it makes more sense to store the discriminant in fastcrypto though and just call a simple interface from the blockchain.

Reply:
I've updated the PR such that it is not possible to create a discriminant without verifying it. I've also added a fixed precomputed discriminant of the size, I expect we're going to use on Sui, but this will have to be updated to a discriminant generated from a seed, we don't control.